crates/data-generator/src/blocks/mod.rs

```diff
@@ -0,0 +1,28 @@
+use std::io::{Error as IoError, Write as IoWrite};
+
+pub(crate) use json::ParseError as JsonParseError;
+use thiserror::Error;
+
+mod json;
+mod model;
+mod output;
+
+#[derive(Debug, Error)]
+pub enum GenerateError {
+    #[error("I/O operation failed")]
+    Io(#[from] IoError),
+    #[error("JSON parsing failed")]
+    Parse(#[from] JsonParseError),
+}
+
+pub(crate) fn gen_blocks_module(
```

Comment:
```suggestion
pub fn gen_blocks_module(
```
Unnecessary encapsulation